kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/ClusterEventTracking.java

```diff
@@ -401,38 +401,38 @@ protected void writeBatchEvents(Collection<Event> events)
 		try
 		{
 			conn = sqlService().borrowConnection();
-			wasCommit = conn.getAutoCommit();
-			if (wasCommit)
-			{
-				conn.setAutoCommit(false);
-			}
 
-			// Note: investigate batch writing via the jdbc driver: make sure we can still use prepared statements (check out host arrays, too)
-			// -ggolden
+			// common preparation for each insert
+			String statement = insertStatement();
 
-            // common preparation for each insert
-            String statement = insertStatement();
-            Object fields[] = new Object[6];
+			// Setup a batch of events if not using a cluster
+			List<Object[]> eventList = new ArrayList<>();
 
 			// write all events
 			for (Event event : events)
 			{
+				Object fields[] = new Object[6];
 				bindValues(event, fields);
+				eventList.add(fields);
+
+				// For clustered setups, going to use legacy, individual inserts
```

Comment:
I find the comment slightly misleading as you can have a clustered setup without caching.